File: go/mysql.go
package mysqldb

import (
	"database/sql"
	"fmt"
	"os"

	_ "github.com/go-sql-driver/mysql"
)

type MySQLDB struct {
	Conn *sql.DB
}

func Connect() (*MySQLDB, error) {
	user := os.Getenv("MYSQL_USER")
	password := os.Getenv("MYSQL_PASS")
	host := os.Getenv("MYSQL_HOST")
	port := os.Getenv("MYSQL_PORT")
	database := os.Getenv("MYSQL_DB")

	dsn := fmt.Sprintf("%s:%s@tcp(%s:%s)/%s", user, password, host, port, database)

	conn, err := sql.Open("mysql", dsn)
	if err != nil {
		return nil, err
	}

	db := &MySQLDB{Conn: conn}
	return db, nil
}

func (db *MySQLDB) Insert(table, query string, args ...interface{}) (sql.Result, error) {
	stmt, err := db.Conn.Prepare(fmt.Sprintf("INSERT INTO %s %s", table, query))
	if err != nil {
		return nil, err
	}
	defer stmt.Close()

	return stmt.Exec(args...)
}

func (db *MySQLDB) Query(query string, args ...interface{}) (*sql.Rows, error) {
	return db.Conn.Query(query, args...)
}

func (db *MySQLDB) Update(query string, args ...interface{}) (sql.Result, error) {
	return db.Conn.Exec(query, args...)
}

func (db *MySQLDB) Delete(query string, args ...interface{}) (sql.Result, error) {
	return db.Conn.Exec(query, args...)
}

func (db *MySQLDB) Close() error {
	return db.Conn.Close()
}

File: python/mysql.py
import os
import mysql.connector

class MySQLDB:
    def __init__(self):
        # Get MySQL connection details from ENV variables
        self.host = os.getenv("MYSQL_HOST", "localhost")
        self.port = int(os.getenv("MYSQL_PORT", 3306))
        self.user = os.getenv("MYSQL_USER", "root")
        self.password = os.getenv("MYSQL_PASS", "")
        self.database = os.getenv("MYSQL_DB", "")

        self.connection = self._connect()
        self.cursor = self.connection.cursor()

    def _connect(self):
        # Establish a connection to the MySQL server
        connection = mysql.connector.connect(
            host=self.host,
            port=self.port,
            user=self.user,
            password=self.password,
            database=self.database
        )
        return connection
    
    def execute_query(self, query, values=None):
        # Execute a SQL query
        if values:
            self.cursor.execute(query, values)
        else:
            self.cursor.execute(query)
        self.connection.commit()

    def fetch_all(self):
        # Fetch all results from the last query
        return self.cursor.fetchall()

    def create_database(self, db_name):
        # Create a new database
        self.execute_query(f"CREATE DATABASE {db_name}")

    def delete_database(self, db_name):
        # Delete the specified database
        self.execute_query(f"DROP DATABASE {db_name}")

    def create_table(self, table_query):
        # Create a new table
        self.execute_query(table_query)

    def delete_table(self, table_name):
        # Delete the specified table
        self.execute_query(f"DROP TABLE {table_name}")

    def insert_record(self, table_name, columns, values):
        # Insert a new record into the table
        query = f"INSERT INTO {table_name} ({', '.join(columns)}) VALUES ({', '.join(['%s' for _ in values])})"
        self.execute_query(query, values)

    def get_records(self, table_name, where_clause=None):
        # Retrieve records from the table
        query = f"SELECT * FROM {table_name}"
        if where_clause:
            query += f" WHERE {where_clause}"
        self.execute_query(query)
        return self.fetch_all()

    def update_record(self, table_name, set_clause, where_clause):
        # Update a record in the table
        query = f"UPDATE {table_name} SET {set_clause} WHERE {where_clause}"
        self.execute_query(query)

    def delete_record(self, table_name, where_clause):
        # Delete a record from the table
        query = f"DELETE FROM {table_name} WHERE {where_clause}"
        self.execute_query(query)

    def close(self):
        # Close the connection
        self.cursor.close()
        self.connection.close()



# import os

# os.environ["MYSQL_HOST"] = "localhost"
# os.environ["MYSQL_PORT"] = "3306"
# os.environ["MYSQL_USER"] = "root"
# os.environ["MYSQL_PASS"] = "password"
# os.environ["MYSQL_DB"] = "testdb"


# db = MySQLDB()

# # Create a sample table
# create_table_query = """
# CREATE TABLE users (
#     id INT AUTO_INCREMENT PRIMARY KEY,
#     username VARCHAR(255) NOT NULL,
#     age INT
# )
# """
# db.create_table(create_table_query)

# # Insert a sample record
# db.insert_record("users", ["username", "age"], ["john", 30])

# # Retrieve records
# print(db.get_records("users"))

# # Update a record
# db.update_record("users", "age=31", "username='john'")

# # Delete a record
# db.delete_record("users", "username='john'")

# db.close()
